File: src/test/java/ar/edu/unq/unidad3/dao/helper/dao/HibernateDataDAO.java
package ar.edu.unq.unidad3.dao.helper.dao;

import ar.edu.unq.unidad3.service.runner.HibernateTransactionRunner;
import org.hibernate.Session;

import java.util.List;

public class HibernateDataDAO implements DataDAO {

    @Override
    public void clear() {
        Session session = HibernateTransactionRunner.getCurrentSession();
        List<?> nombreDeTablas = session.createNativeQuery("show tables").getResultList();
        session.createNativeQuery("SET FOREIGN_KEY_CHECKS=0;").executeUpdate();
        for (Object result : nombreDeTablas) {
            String tabla = "";
            if (result instanceof String) {
                tabla = (String) result;
            } else if (result instanceof Object[]) {
                tabla = ((Object[]) result)[0].toString();
            }
            session.createNativeQuery("truncate table " + tabla).executeUpdate();
        }
        session.createNativeQuery("SET FOREIGN_KEY_CHECKS=1;").executeUpdate();
    }
}

File: src/main/kotlin/ar/edu/unq/unidad3/dao/impl/HibernateDataDAO.kt
package ar.edu.unq.unidad3.dao.impl

import ar.edu.unq.unidad3.dao.DataDAO
import ar.edu.unq.unidad3.service.runner.TransactionRunner
import org.hibernate.Session

open class HibernateDataDAO : DataDAO {

    override fun clear() {
        val session = TransactionRunner.currentSession
        val nombreDeTablas = session.createNativeQuery("show tables").resultList
        session.createNativeQuery("SET FOREIGN_KEY_CHECKS=0;").executeUpdate()
        nombreDeTablas.forEach { tabla -> session.createNativeQuery("truncate table $tabla").executeUpdate() }
        session.createNativeQuery("SET FOREIGN_KEY_CHECKS=1;").executeUpdate()
    }
}
